backend/src/fhir_repository.py

```diff
@@ -365,43 +350,39 @@ def _perform_dynamo_update(
                 )
 
     def delete_immunization(
-        self, imms_id: str, imms_vax_type_perms: str, supplier_system: str, is_imms_batch_app
-    ) -> dict:
+            self, imms_id: str, imms_vax_type_perms: str, supplier_system: str) -> dict:
         now_timestamp = int(time.time())
+        
         try:
-            resp = self.table.get_item(Key={"PK": _make_immunization_pk(imms_id)})
-
-            if "Item" in resp:
-                if "DeletedAt" in resp["Item"]:
-                    if resp["Item"]["DeletedAt"] == "reinstated":
-                        vaccine_type = self._vaccine_type(resp["Item"]["PatientSK"])
-                        if not is_imms_batch_app:
-                            vax_type_perms = self._parse_vaccine_permissions(imms_vax_type_perms)
-                            vax_type_perm = self._vaccine_permission(vaccine_type, "delete")
-                            self._check_permission(vax_type_perm, vax_type_perms)
-                else:
-                    vaccine_type = self._vaccine_type(resp["Item"]["PatientSK"])
-                    if not is_imms_batch_app:
-                        vax_type_perms = self._parse_vaccine_permissions(imms_vax_type_perms)
-                        vax_type_perm = self._vaccine_permission(vaccine_type, "delete")
-                        self._check_permission(vax_type_perm, vax_type_perms)
+            item = self.table.get_item(Key={"PK": _make_immunization_pk(imms_id)}).get("Item")
+            if not item:
+                raise ResourceNotFoundError(resource_type="Immunization", resource_id=imms_id)
+            
+            if item.get("DeletedAt") == "reinstated":
```

Comment:
I think this needs to be:
```suggestion
            if not item.get("DeletedAt") or item.get("DeletedAt") == "reinstated":
```
so that we check permissions for non-deleted items.